modules/aws-sgw/main.tf

```diff
@@ -2,17 +2,31 @@
 # Storage Gateway
 ################################################################################
 
+locals {
+  create_smb_active_directory_settings = (var.join_smb_domain == true && length(var.domain_controllers) > 0 && length(var.domain_name) > 0 && length(var.domain_password) > 0 && length(var.domain_username) > 0)
```

Comment:
just nit pick
breaking this into smaller locals for each validation, and then combined it would help from readability standpoint.